bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/helper/image/AssetDeliveryHelper.java

```diff
@@ -74,8 +71,33 @@ public static String getSrcSet(@NotNull AssetDelivery assetDelivery, @NotNull Re
         return null;
     }
 
-    public static  String getSrc(@NotNull AssetDelivery assetDelivery, @NotNull Resource imageComponentResource, @NotNull String imageName,
-                                 @NotNull String extension, @NotNull int[] smartSizes, int jpegQuality) {
+    public static String getSrcUriTemplate(@NotNull AssetDelivery assetDelivery, @NotNull Resource imageComponentResource,
+                                           @NotNull String imageName, @NotNull String extension,
+                                           @NotNull int[] smartSizes, int jpegQuality, String widthPlaceholder) {
```

Comment:
The more a look at it, the more if find passing the `int[] smartSizes` to not really be necessary. Instead we could have an optional width as all methods (`getSrc`, `getSrcUriTemplate`) actually return just only one String

Reply:
`smartSizes` cannot be an optional because its length decides whether we add `quality` in the URL or not.